apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -121,7 +124,15 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
         /// </summary>
```

Comment:
Yea that works for me.  This works as is if IManagedTracer is used in a scoped manner.

I think the other issue we are seeing is an issue with the propagation of the headers.